amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/declaration/Raml10TypeParser.scala

```diff
@@ -317,7 +317,7 @@ case class Raml08UnionTypeParser(shape: UnionShape, types: Seq[YNode], ast: YPar
       .filter(_.isDefined)
       .map(_.get)
 
-    shape.setArray(UnionShapeModel.AnyOf, unionNodes, Annotations(ast))
```

Comment:
Not necessary. Some client methods can set just values in a memory model. In those cases, setArray maybe could be useful.
We should not use setArray in parsing clases. Another approach could be force the setArray method to build the AMfArray with virtual element annotation, but it could be weird for the client building model use case.